src/compiler/checker.ts

```diff
@@ -167,10 +167,10 @@ namespace ts {
 
         return checker;
 
-        function getEmitResolver(sourceFile?: SourceFile) {
+        function getEmitResolver(sourceFile: SourceFile, cancellationToken: CancellationToken) {

```

Comment:
but this is the model we have used thus far everywhere else. we do not create new cancellation tokens per operation. we relay on the fact that the JS engine is single threaded. so we have at most one operation at a time. 
We should not have two ways of doing the same thing.